zerobuf/ConstAllocator.h

```diff
@@ -0,0 +1,43 @@
+
+/* Copyright (c) 2015-2016, Human Brain Project
+ *                          Daniel.Nachbaur@epfl.ch
+ */
+
+#ifndef ZEROBUF_CONSTALLOCATOR_H
+#define ZEROBUF_CONSTALLOCATOR_H
+
+#include <zerobuf/api.h>
+#include <zerobuf/Allocator.h> // base class
+
+namespace zerobuf
+{
+/** A zerobuf root allocator which gives read-only access to its data. */
+class ConstAllocator : public Allocator
+{
+public:
+    ZEROBUF_API ConstAllocator( const uint8_t* data, size_t size );
+
+    ZEROBUF_API ~ConstAllocator();
+
+    const uint8_t* getData() const final { return _data; }
+    size_t getSize() const final { return _size; }
+
+private:
+    ConstAllocator( const ConstAllocator& ) = delete;
+    ConstAllocator& operator = ( const ConstAllocator& ) = delete;
+
+    uint8_t* getData() final
+        { return const_cast< uint8_t* >( _data ); } // internal
```

Comment:
I had the throw before for the same reasons, but it does not work unfortunately. With the test schema in the unit tests, it passes by code which calls getData() and the compiler prefers the non-const over the const version, hence it passes by here. A delete for this non-const getData() is what one wants here...